python_modules/dagster/dagster/_core/pipes/context.py

```diff
@@ -427,6 +446,18 @@ def get_custom_messages(self) -> Sequence[Any]:
         """
         return self.message_handler.get_custom_messages()
 
+    def report_launched(self, launched_payload: PipesLaunchedData):
+        """Submit arbitrary information about the launched process to Pipes.
+
+        This hook is not necessarily called in every pipes session. It is useful primarily when we wish to
+        condition the behavior of Pipes on some parameter that is only available after
+        external process launch (such as a run id in the external system). The code calling `open_pipes_session()`
+        is responsible for calling `PipesSession.report_launched()`.
+
+        Passes the `launched_payload` to the message reader's `on_launched` method.
+        """
+        self.message_handler._message_reader.on_launched(launched_payload)  # noqa: SLF001
```

Comment:
Let's route this through the `message_handler` (you can add an `on_launched` method to it that just calls `MessageReader.on_launched()` instead of private attribute access.